CMake cleanup

axel-h opened the pull request:
removal of KernelArmSel4Arch usage here is needed for https://github.com/seL4/seL4/pull/274

axel-h commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
All the tests we ran have passed! Nice job!

kent-mcleod commented on cmake-tool/helpers/application_settings.cmake:
```diff
@@ -96,79 +96,93 @@ endfunction()
+        "imx6,sabre,wandq"
+        "imx31,kzm"
+        "bcm2837,rpi3"
+        "exynos5,exynos5250,exynos5410,exynos5,exynos5422,exynos5"
```
Are you able to add a comment describing the new format of this list?
I'm a bit confused about how to interpret the exynos line.

axel-h replied:
Thanks for the review, all "exynos5" except the first one are to be removed. That's a left over from an earlier attempt. Will add a comment describing the format.

axel-h commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
All the tests we ran have passed! Nice job!

yyshen commented on cmake-tool/helpers/application_settings.cmake:
```diff
@@ -96,89 +91,181 @@ endfunction()
+        "bcm2837,rpi3"
+        "exynos5,exynos5250,exynos5410,exynos5422"
+        "am335x,am335x-boneblack,am335x-boneblue"
+        "-KernelSel4Arch"
```
just curious, where are the aarch64 platforms (tx1, tx2, hikey ...)?

axel-h replied:
I've just translated into this "structure" what was used in the CAmke code before - which was `KernelSel4Arch`. For RISC-V, we don't have anything yet, because there was no need. We can do further refactoring based on this now, but I'd prefer to merge this first to have a baseline.

axel-h replied:
@yyshen: again, I just translated what existed there before. Just not with the cleaner approach is becomes more obvious what happens here and what does not happen here. See the comment about this structure, there is says
```
If there was no match for any board the function will set:
     KernelPlatform=${PLATFORM}
 and leave all further setup to the architecture/platform specific
 configuration to <sel4 kernel>/src/plat/*/config.cmake
```
So what will happen is, that these file than act upon the setting of KernelPlatform. Either they have only one platform anyway or they have a fallback to a default board then. 
The whole things with the platform is still a bit of a mess and I think `correct_platform_strings()` only exists as an attempt to clean this up and address some issues that are eventually to be fixed in the `<sel4 kernel>/src/plat/*/config.cmake` files.  With my cleanup, it should get easier to get this started, as it becomes more clear now, what actually happens.

kent-mcleod replied:
This was originally done to avoid having to propagate platform name changes through all of the CI scripts that would refer to platforms by different names, such as rpi and bcm2837.  I think the inspiration of making aliases came from seeing what netbsd does: https://github.com/IIJ-NetBSD/netbsd-src/blob/master/build.sh#L618

axel-h commented on cmake-tool/helpers/application_settings.cmake:
```diff
@@ -96,89 +91,181 @@ endfunction()
+            string(
+                REPLACE
+                    ","
+                    ";" # makes string a CMake list
+                    plat_filter
+                    "${item}"
+            )
+            list(GET plat_filter 0 plat_target)
+            list(REMOVE_AT plat_filter 0)
+
+            if("${PLATFORM}" IN_LIST plat_filter)
+
+                if(
+                    KernelPlatform
+                    AND
+                        NOT
+                        "${KernelPlatform}"
+                        STREQUAL
+                        "${plat_target}"
+                )
+                    message(
+                        FATAL_ERROR
+                            "config mismatch, wont overwrite KernelPlatform=${KernelPlatform} with ${plat_target}"
+                    )
+                endif()
+                set(KernelPlatform "${plat_target}" CACHE STRING "" FORCE)
+
+                if(
+                    ${kernel_var}
+                    AND
+                        NOT
+                        "${${kernel_var}}"
+                        STREQUAL
+                        "${PLATFORM}"
+                )
+                    message(
+                        FATAL_ERROR
+                            "config mismatch, wont overwrite ${kernel_var}=${${kernel_var}} with ${PLATFORM}"
+                    )
+                endif()
```
which indent? I had some fights with the style check, so to seem what it wants.

szhuang replied:
I ran the style checker at my end. It has a strange taste. But yeah, this is what it wants.

axel-h commented on cmake-tool/helpers/application_settings.cmake:
```diff
@@ -96,89 +91,181 @@ endfunction()
+            string(
+                REPLACE
+                    ","
+                    ";" # makes string a CMake list
```
According to https://cmake.org/cmake/help/latest/command/set.html "multiple arguments will be joined as a semicolon-separated list", so this is what happens happens for the elements in `correct_platform_strings_platform_aliases`. However, I need a "list-in-list", thus I use "," as a separator for the 2nd layer list and by "," with ";" and can accesses the 2nd layer list. This is a common CMake practice, any separator will so there.

walle-whale commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
## Test results for commit 665e00aee70171da9205eb326eaba8e5028f393a

### Summary

#### NOTE: some test results have been updated!

| Test name | Result |
| --- | --- |
| Code Quality - Check licenses - seL4_tools - EXT-axel-h-patch-axel-3-48 | SUCCESSFUL |
| Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 | SUCCESSFUL |
| Code Quality - Signed-off-by - seL4_tools - EXT-axel-h-patch-axel-3-48 | SUCCESSFUL |
| Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 | FAILED |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 | Style check on changed files | 1 of 3 jobs failed |

**Job: gitlint**
Build logs

    04-Nov-2020 15:36:29	Build Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 - gitlint #3 (CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR-3) started building on agent A local agent 2
    04-Nov-2020 15:36:29	I run on the Bamboo server!
    04-Nov-2020 15:36:29	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR
    04-Nov-2020 15:36:29	Executing build Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 - gitlint #3 (CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR-3)
    04-Nov-2020 15:36:29	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR/. (location: .)
    04-Nov-2020 15:36:29	Artifact 'code' prepared in 22.87 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR/./code.tar.gz
    04-Nov-2020 15:36:29	Running pre-build action: VCS Version Collector
    04-Nov-2020 15:36:29	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 15:36:29	+ tar xf code.tar.gz --recursive-unlink
    04-Nov-2020 15:36:29	Finished task 'Untar code' with result: Success
    04-Nov-2020 15:36:29	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 15:36:29	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR
    04-Nov-2020 15:36:29	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    04-Nov-2020 15:36:29	Starting task 'Lint commit messages' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 15:36:29	+ cd seL4_tools
    04-Nov-2020 15:36:29	+ '[' -f .gitlint ']'
    04-Nov-2020 15:36:29	+ CONFIG=/var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR/seL4_tools/misc/.gitlint
    04-Nov-2020 15:36:29	+ gitlint --commits 31d847ce5f599c139c51f9a22843b0e2898ba4c5..665e00aee70171da9205eb326eaba8e5028f393a --config /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR/seL4_tools/misc/.gitlint
    04-Nov-2020 15:36:45	Commit 665e00aee7:
    04-Nov-2020 15:36:45	3: B1 Line exceeds max length (78>72): "This aligns it with other message from the build system, that also use STATUS."
    04-Nov-2020 15:36:45	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR-3-ScriptBuildTask-2375686567375811881.sh] was 1 while expected 0
    04-Nov-2020 15:36:45	Finished task 'Lint commit messages' with result: Failed
    04-Nov-2020 15:36:45	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 15:36:45	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR
    04-Nov-2020 15:36:45	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    04-Nov-2020 15:36:45	Running post build plugin 'NCover Results Collector'
    04-Nov-2020 15:36:45	Running post build plugin 'Artifact Copier'
    04-Nov-2020 15:36:45	Running post build plugin 'npm Cache Cleanup'
    04-Nov-2020 15:36:45	Running post build plugin 'Clover Results Collector'
    04-Nov-2020 15:36:45	Running post build plugin 'Docker Container Cleanup'
    04-Nov-2020 15:36:45	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR'
    04-Nov-2020 15:36:45	Finalising the build...
    04-Nov-2020 15:36:45	Stopping timer.
    04-Nov-2020 15:36:45	Build CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR-3 completed.
    04-Nov-2020 15:36:45	Running on server: post build plugin 'NCover Results Collector'
    04-Nov-2020 15:36:45	Running on server: post build plugin 'Build Hanging Detection Configuration'
    04-Nov-2020 15:36:45	Running on server: post build plugin 'Clover Delta Calculator'
    04-Nov-2020 15:36:45	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    04-Nov-2020 15:36:45	All post build plugins have finished
    04-Nov-2020 15:36:45	Generating build results summary...
    04-Nov-2020 15:36:45	Saving build results to disk...
    04-Nov-2020 15:36:45	Store variable context...
    04-Nov-2020 15:36:45	Indexing build results...
    04-Nov-2020 15:36:45	Finished building CQ-SEL4TOOLSSTYLE162-SEL4TOOLSGITLINTPR-3.


This is the most I can report on right now, sorry!

szhuang commented:
All good, I'll mark it internally.

axel-h commented:
fixed the comment line length

axel-h commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
## Test results for commit c78eadb1bd21d10a3fc06caedb478c379f2f8dea

### Summary

#### NOTE: some test results have been updated!

| Test name | Result |
| --- | --- |
| Code Quality - Check licenses - seL4_tools - EXT-axel-h-patch-axel-3-48 | SUCCESSFUL |
| Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 | FAILED |
| Code Quality - Signed-off-by - seL4_tools - EXT-axel-h-patch-axel-3-48 | SUCCESSFUL |
| Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 | FAILED |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 | Reuse License Check | 1 of 1 job failed |

**Job: Reuse License Check**
Build logs

    04-Nov-2020 22:17:19	Build Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 - Reuse License Check #4 (CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-4) started building on agent A local agent 2
    04-Nov-2020 22:17:19	I run on the Bamboo server!
    04-Nov-2020 22:17:19	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ
    04-Nov-2020 22:17:19	Executing build Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 - Reuse License Check #4 (CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-4)
    04-Nov-2020 22:17:19	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ/. (location: .)
    04-Nov-2020 22:17:20	Artifact 'code' prepared in 707.9 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ/./code.tar.gz
    04-Nov-2020 22:17:20	Running pre-build action: VCS Version Collector
    04-Nov-2020 22:17:20	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:20	+ tar xf code.tar.gz --recursive-unlink
    04-Nov-2020 22:17:20	Finished task 'Untar code' with result: Success
    04-Nov-2020 22:17:20	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:20	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ
    04-Nov-2020 22:17:20	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    04-Nov-2020 22:17:20	Starting task 'Reuse License Check' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:20	+ cd seL4_tools
    04-Nov-2020 22:17:20	+ '[' -L isabelle ']'
    04-Nov-2020 22:17:20	+ reuse --root . lint
    04-Nov-2020 22:17:20	/var/atlassian/application-data/bamboo/temp/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-4-ScriptBuildTask-6763669096322142395.sh: line 8: reuse: command not found
    04-Nov-2020 22:17:20	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-4-ScriptBuildTask-6763669096322142395.sh] was 127 while expected 0
    04-Nov-2020 22:17:20	Finished task 'Reuse License Check' with result: Failed
    04-Nov-2020 22:17:20	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:20	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ
    04-Nov-2020 22:17:20	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    04-Nov-2020 22:17:20	Running post build plugin 'NCover Results Collector'
    04-Nov-2020 22:17:20	Running post build plugin 'Artifact Copier'
    04-Nov-2020 22:17:20	Running post build plugin 'npm Cache Cleanup'
    04-Nov-2020 22:17:20	Running post build plugin 'Clover Results Collector'
    04-Nov-2020 22:17:20	Running post build plugin 'Docker Container Cleanup'
    04-Nov-2020 22:17:20	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/151224373/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ'
    04-Nov-2020 22:17:20	Finalising the build...
    04-Nov-2020 22:17:20	Stopping timer.
    04-Nov-2020 22:17:20	Build CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-4 completed.
    04-Nov-2020 22:17:20	Running on server: post build plugin 'NCover Results Collector'
    04-Nov-2020 22:17:20	Running on server: post build plugin 'Build Hanging Detection Configuration'
    04-Nov-2020 22:17:20	Running on server: post build plugin 'Clover Delta Calculator'
    04-Nov-2020 22:17:20	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    04-Nov-2020 22:17:20	All post build plugins have finished
    04-Nov-2020 22:17:20	Generating build results summary...
    04-Nov-2020 22:17:20	Saving build results to disk...
    04-Nov-2020 22:17:20	Store variable context...
    04-Nov-2020 22:17:20	Indexing build results...
    04-Nov-2020 22:17:20	Finished building CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-4.


| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 | Style check on changed files | 1 of 3 jobs failed |

**Job: style**
Build logs

    04-Nov-2020 22:17:22	Build Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 - style #4 (CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR-4) started building on agent A local agent 1
    04-Nov-2020 22:17:22	I run on the Bamboo server!
    04-Nov-2020 22:17:22	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR
    04-Nov-2020 22:17:22	Executing build Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 - style #4 (CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR-4)
    04-Nov-2020 22:17:22	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR/. (location: .)
    04-Nov-2020 22:17:22	Artifact 'code' prepared in 33.70 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR/./code.tar.gz
    04-Nov-2020 22:17:22	Running pre-build action: VCS Version Collector
    04-Nov-2020 22:17:22	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:22	+ tar xf code.tar.gz --recursive-unlink
    04-Nov-2020 22:17:22	Finished task 'Untar code' with result: Success
    04-Nov-2020 22:17:22	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:22	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR
    04-Nov-2020 22:17:22	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    04-Nov-2020 22:17:22	Starting task 'Check style of changed files' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:22	+ cd seL4_tools
    04-Nov-2020 22:17:22	+ xargs -d '\n' /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR/standard_seL4_tools/misc/style.sh
    04-Nov-2020 22:17:22	++ git merge-base c78eadb1bd21d10a3fc06caedb478c379f2f8dea 31d847ce5f599c139c51f9a22843b0e2898ba4c5
    04-Nov-2020 22:17:22	+ git diff --name-only c78eadb1bd21d10a3fc06caedb478c379f2f8dea 31d847ce5f599c139c51f9a22843b0e2898ba4c5
    04-Nov-2020 22:17:23	+ git diff --exit-code
    04-Nov-2020 22:17:23	diff --git a/cmake-tool/helpers/application_settings.cmake b/cmake-tool/helpers/application_settings.cmake
    04-Nov-2020 22:17:23	index 9f3c9c6..97c0642 100644
    04-Nov-2020 22:17:23	--- a/cmake-tool/helpers/application_settings.cmake
    04-Nov-2020 22:17:23	+++ b/cmake-tool/helpers/application_settings.cmake
    04-Nov-2020 22:17:23	@@ -270,10 +270,7 @@ function(correct_platform_strings)
    04-Nov-2020 22:17:23	         set(_REWRITE OFF)
    04-Nov-2020 22:17:23	     endif()
    04-Nov-2020 22:17:23	     if(_REWRITE AND (NOT correct_platform_strings_no_print))
    04-Nov-2020 22:17:23	-        message(
    04-Nov-2020 22:17:23	-            STATUS
    04-Nov-2020 22:17:23	-                "Setting from flags KernelSel4Arch: ${KernelSel4Arch}"
    04-Nov-2020 22:17:23	-        )
    04-Nov-2020 22:17:23	+        message(STATUS "Setting from flags KernelSel4Arch: ${KernelSel4Arch}")
    04-Nov-2020 22:17:23	     endif()
    04-Nov-2020 22:17:23	 
    04-Nov-2020 22:17:23	     # Only print out these info messages on first initialisation
    04-Nov-2020 22:17:23	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR-4-ScriptBuildTask-2642996306166524461.sh] was 1 while expected 0
    04-Nov-2020 22:17:23	Finished task 'Check style of changed files' with result: Failed
    04-Nov-2020 22:17:23	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:23	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR
    04-Nov-2020 22:17:23	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    04-Nov-2020 22:17:23	Starting task 'Run astyle code insights' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:17:23	+ cd seL4_tools
    04-Nov-2020 22:17:23	++ cut -d/ -f4
    04-Nov-2020 22:17:23	++ echo ssh://git@bitbucket.ts.data61.csiro.au:7999/sel4/sel4_tools.git
    04-Nov-2020 22:17:23	+ bb_proj=sel4
    04-Nov-2020 22:17:23	+ /scripts/bb_code_insights/create_code_insights.sh astyle sel4 seL4_tools
    04-Nov-2020 22:17:23	++ dirname /scripts/bb_code_insights/create_code_insights.sh
    04-Nov-2020 22:17:23	+ SCRIPT_DIR=/scripts/bb_code_insights
    04-Nov-2020 22:17:23	+ BBS_URL=https://bitbucket.ts.data61.csiro.au
    04-Nov-2020 22:17:23	+ BBS_PROJECT=sel4
    04-Nov-2020 22:17:23	+ BBS_REPO=seL4_tools
    04-Nov-2020 22:17:23	+ REPORT_TO_RUN=astyle
    04-Nov-2020 22:17:23	++ git rev-parse HEAD
    04-Nov-2020 22:17:23	+ COMMIT_ID=c78eadb1bd21d10a3fc06caedb478c379f2f8dea
    04-Nov-2020 22:17:23	+ REPORT_KEY=astyle.report
    04-Nov-2020 22:17:23	+ set +x
    04-Nov-2020 22:17:23	+ report_file=astyle_report.json
    04-Nov-2020 22:17:23	+ annotations_file=astyle_annotations.json
    04-Nov-2020 22:17:23	+ rm -f astyle_report.json astyle_annotations.json
    04-Nov-2020 22:17:23	+ /scripts/bb_code_insights/astyle
    04-Nov-2020 22:17:25	+ echo 'Reporting the results to BitBucket, but doing so *secretly*, so people don'\''t see the access token'
    04-Nov-2020 22:17:25	Reporting the results to BitBucket, but doing so *secretly*, so people don't see the access token
    04-Nov-2020 22:17:25	+ set +x
    04-Nov-2020 22:17:25	+ '[' -f astyle_report.json ']'
    04-Nov-2020 22:17:25	+ cat astyle_report.json
    04-Nov-2020 22:17:25	+ '[' -f astyle_annotations.json ']'
    04-Nov-2020 22:17:25	{"data":[{"title":"Total number of files with incorrect style","value":0},{"title":"Total number of lines with incorrect style","value":0}],"createdDate":1604488645599,"key":"astyle.report","title":"Code style"}{"title": "Code style", "data": [{"title": "Total number of files with incorrect style", "value": 0}, {"title": "Total number of lines with incorrect style", "value": 0}]}
    04-Nov-2020 22:17:25	Finished task 'Run astyle code insights' with result: Success
    04-Nov-2020 22:17:25	Running post build plugin 'NCover Results Collector'
    04-Nov-2020 22:17:25	Running post build plugin 'Artifact Copier'
    04-Nov-2020 22:17:25	Running post build plugin 'npm Cache Cleanup'
    04-Nov-2020 22:17:25	Running post build plugin 'Clover Results Collector'
    04-Nov-2020 22:17:25	Running post build plugin 'Docker Container Cleanup'
    04-Nov-2020 22:17:25	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR'
    04-Nov-2020 22:17:25	Finalising the build...
    04-Nov-2020 22:17:25	Stopping timer.
    04-Nov-2020 22:17:25	Build CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR-4 completed.
    04-Nov-2020 22:17:25	Running on server: post build plugin 'NCover Results Collector'
    04-Nov-2020 22:17:25	Running on server: post build plugin 'Build Hanging Detection Configuration'
    04-Nov-2020 22:17:25	Running on server: post build plugin 'Clover Delta Calculator'
    04-Nov-2020 22:17:25	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    04-Nov-2020 22:17:25	All post build plugins have finished
    04-Nov-2020 22:17:25	Generating build results summary...
    04-Nov-2020 22:17:25	Saving build results to disk...
    04-Nov-2020 22:17:25	Store variable context...
    04-Nov-2020 22:17:25	Indexing build results...
    04-Nov-2020 22:17:25	Finished building CQ-SEL4TOOLSSTYLE162-SEL4TOOLSSTYLEPR-4.


This is the most I can report on right now, sorry!

axel-h commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
## Test results for commit ff0d0cbbe289be396c4d7c45c3f9db4b883758a5

### Summary

#### NOTE: some test results have been updated!

| Test name | Result |
| --- | --- |
| Code Quality - Check licenses - seL4_tools - EXT-axel-h-patch-axel-3-48 | SUCCESSFUL |
| Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 | FAILED |
| Code Quality - Signed-off-by - seL4_tools - EXT-axel-h-patch-axel-3-48 | SUCCESSFUL |
| Code Quality - Style - seL4_tools - EXT-axel-h-patch-axel-3-48 | SUCCESSFUL |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 | Reuse License Check | 1 of 1 job failed |

**Job: Reuse License Check**
Build logs

    04-Nov-2020 22:24:49	Build Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 - Reuse License Check #5 (CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-5) started building on agent A local agent 1
    04-Nov-2020 22:24:49	I run on the Bamboo server!
    04-Nov-2020 22:24:49	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ
    04-Nov-2020 22:24:49	Executing build Code Quality - Reuse License Check - seL4_tools - EXT-axel-h-patch-axel-3-48 - Reuse License Check #5 (CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-5)
    04-Nov-2020 22:24:49	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ/. (location: .)
    04-Nov-2020 22:24:49	Artifact 'code' prepared in 33.85 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ/./code.tar.gz
    04-Nov-2020 22:24:49	Running pre-build action: VCS Version Collector
    04-Nov-2020 22:24:49	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:24:49	+ tar xf code.tar.gz --recursive-unlink
    04-Nov-2020 22:24:49	Finished task 'Untar code' with result: Success
    04-Nov-2020 22:24:49	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:24:49	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ
    04-Nov-2020 22:24:49	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    04-Nov-2020 22:24:49	Starting task 'Reuse License Check' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:24:49	+ cd seL4_tools
    04-Nov-2020 22:24:49	+ '[' -L isabelle ']'
    04-Nov-2020 22:24:49	+ reuse --root . lint
    04-Nov-2020 22:24:49	/var/atlassian/application-data/bamboo/temp/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-5-ScriptBuildTask-7489716778215958671.sh: line 8: reuse: command not found
    04-Nov-2020 22:24:49	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-5-ScriptBuildTask-7489716778215958671.sh] was 127 while expected 0
    04-Nov-2020 22:24:49	Finished task 'Reuse License Check' with result: Failed
    04-Nov-2020 22:24:49	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    04-Nov-2020 22:24:49	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ
    04-Nov-2020 22:24:49	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    04-Nov-2020 22:24:49	Running post build plugin 'NCover Results Collector'
    04-Nov-2020 22:24:49	Running post build plugin 'Artifact Copier'
    04-Nov-2020 22:24:49	Running post build plugin 'npm Cache Cleanup'
    04-Nov-2020 22:24:49	Running post build plugin 'Clover Results Collector'
    04-Nov-2020 22:24:49	Running post build plugin 'Docker Container Cleanup'
    04-Nov-2020 22:24:49	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ'
    04-Nov-2020 22:24:49	Finalising the build...
    04-Nov-2020 22:24:49	Stopping timer.
    04-Nov-2020 22:24:49	Build CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-5 completed.
    04-Nov-2020 22:24:49	Running on server: post build plugin 'NCover Results Collector'
    04-Nov-2020 22:24:49	Running on server: post build plugin 'Build Hanging Detection Configuration'
    04-Nov-2020 22:24:49	Running on server: post build plugin 'Clover Delta Calculator'
    04-Nov-2020 22:24:49	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    04-Nov-2020 22:24:49	All post build plugins have finished
    04-Nov-2020 22:24:49	Generating build results summary...
    04-Nov-2020 22:24:49	Saving build results to disk...
    04-Nov-2020 22:24:49	Store variable context...
    04-Nov-2020 22:24:49	Indexing build results...
    04-Nov-2020 22:24:50	Finished building CQ-SEL4TOOLSREUSE15-SEL4TOOLSREUSELICENSECHECKCQ-5.


This is the most I can report on right now, sorry!

axel-h commented:
Not sure what to make of this Bamboo error now.

lsf37 commented:
`line 8: reuse: command not found` looks like the bamboo container doesn't have the reuse command any more. The same test is running on github, so if that's the only failing thing, it's fine to ignore for this patch.

@LukeMondy did we rebuild containers recently and changed something there? There should be a `pip3 install reuse` somewhere in the chain.

ssrg-bamboo commented:
All the tests we ran have passed! Nice job!

axel-h commented:
update; using brackets convinces the style checker not to insist on putting everything on its own line in CMake if-statements with multiple conditions

ssrg-bamboo commented:
All the tests we ran have passed! Nice job!

oliver-wm commented:
@ssrg-bamboo `test`

LukeMondy commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
All the tests we ran have passed! Nice job!

axel-h commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
All the tests we ran have passed! Nice job!

oliver-wm commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
All the tests we ran have passed! Nice job!

axel-h commented:
Can you merge this. Then the build error in https://github.com/seL4/seL4/pull/274 will also disappear.

axel-h commented:
I've reworked seL4/seL4#274 and removed the changes that changes that depend on this (will bring in this later). So it there a chance to merge this now?

axel-h commented:
@ssrg-bamboo `test`

ssrg-bamboo commented:
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

ssrg-bamboo commented:
All the tests we ran have passed! Nice job!
